### ISSUE TITLE
Failed to unregister command with commodore

### ISSUE THREAD

**Issue by 0dinD** [Incendo/cloud#388]
Preconditions:
- Use `cloud-bukkit` or `cloud-paper` with `commodore` >= 2.1
    - For example, Paper 1.14.4 or Spigot 1.18.2
- Call `BukkitCommandManager#registerBrigadier()`
- Start the server with the plugin, and then stop the server (to trigger command unregistration)

This causes the following error in the logs:

<details>

<summary>Paper 1.14.4</summary>

```
java.lang.RuntimeException: Failed to unregister command 'uhccore:uhccore' with commodore
        at cloud.commandframework.bukkit.CloudCommodoreManager.unregisterWithCommodore(CloudCommodoreManager.java:145) ~[?:?]
        at cloud.commandframework.bukkit.CloudCommodoreManager.unregisterExternal(CloudCommodoreManager.java:86) ~[?:?]
        at java.lang.Iterable.forEach(Iterable.java:75) ~[?:1.8.0_332]
        at cloud.commandframework.bukkit.BukkitPluginRegistrationHandler.unregisterRootCommand(BukkitPluginRegistrationHandler.java:163) ~[?:?]
        at cloud.commandframework.CommandManager.deleteRootCommand(CommandManager.java:545) ~[?:?]
        at java.util.ArrayList.forEach(ArrayList.java:1259) ~[?:1.8.0_332]
        at cloud.commandframework.bukkit.CloudBukkitListener.onPluginDisable(CloudBukkitListener.java:54) ~[?:?]
        at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:37) ~[patched_1.14.4.jar:git-Paper-245]
        at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:76) ~[patched_1.14.4.jar:git-Paper-245]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.14.4.jar:git-Paper-245]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:545) ~[patched_1.14.4.jar:git-Paper-245]
        at org.bukkit.plugin.java.JavaPluginLoader.disablePlugin(JavaPluginLoader.java:367) ~[patched_1.14.4.jar:git-Paper-245]
        at org.bukkit.plugin.SimplePluginManager.disablePlugin(SimplePluginManager.java:453) ~[patched_1.14.4.jar:git-Paper-245]
        at org.bukkit.plugin.SimplePluginManager.disablePlugins(SimplePluginManager.java:439) ~[patched_1.14.4.jar:git-Paper-245]
        at org.bukkit.plugin.SimplePluginManager.disablePlugins(SimplePluginManager.java:432) ~[patched_1.14.4.jar:git-Paper-245]
        at org.bukkit.craftbukkit.v1_14_R1.CraftServer.disablePlugins(CraftServer.java:401) ~[patched_1.14.4.jar:git-Paper-245]
        at net.minecraft.server.v1_14_R1.MinecraftServer.stop(MinecraftServer.java:733) ~[patched_1.14.4.jar:git-Paper-245]
        at net.minecraft.server.v1_14_R1.DedicatedServer.stop(DedicatedServer.java:739) ~[patched_1.14.4.jar:git-Paper-245]
        at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:966) ~[patched_1.14.4.jar:git-Paper-245]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_332]
Caused by: java.lang.NoSuchMethodException: me.lucko.commodore.ReflectionCommodore.removeChild(com.mojang.brigadier.tree.RootCommandNode, java.lang.String)
        at java.lang.Class.getDeclaredMethod(Class.java:2130) ~[?:1.8.0_332]
        at cloud.commandframework.bukkit.CloudCommodoreManager.unregisterWithCommodore(CloudCommodoreManager.java:130) ~[?:?]
        ... 19 more
```

</details>

<details>

<summary>Spigot 1.18.2</summary>

```
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:310) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.java.JavaPluginLoader.disablePlugin(JavaPluginLoader.java:361) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.disablePlugin(SimplePluginManager.java:501) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.plugin.SimplePluginManager.disablePlugins(SimplePluginManager.java:493) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
	at org.bukkit.craftbukkit.v1_18_R2.CraftServer.disablePlugins(CraftServer.java:452) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3498-Spigot-b081915-10ba1be]
	at net.minecraft.server.MinecraftServer.t(MinecraftServer.java:899) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3498-Spigot-b081915-10ba1be]
	at net.minecraft.server.dedicated.DedicatedServer.t(DedicatedServer.java:717) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3498-Spigot-b081915-10ba1be]
	at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1086) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3498-Spigot-b081915-10ba1be]
	at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:304) ~[spigot-1.18.2-R0.1-SNAPSHOT.jar:3498-Spigot-b081915-10ba1be]
	at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: java.lang.RuntimeException: Failed to unregister command 'uhccore:uhccore' with commodore
	at cloud.commandframework.bukkit.CloudCommodoreManager.unregisterWithCommodore(CloudCommodoreManager.java:145) ~[?:?]
	at cloud.commandframework.bukkit.CloudCommodoreManager.unregisterExternal(CloudCommodoreManager.java:86) ~[?:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at cloud.commandframework.bukkit.BukkitPluginRegistrationHandler.unregisterRootCommand(BukkitPluginRegistrationHandler.java:163) ~[?:?]
	at cloud.commandframework.CommandManager.deleteRootCommand(CommandManager.java:545) ~[?:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at cloud.commandframework.bukkit.CloudBukkitListener.onPluginDisable(CloudBukkitListener.java:54) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
	... 12 more
Caused by: java.lang.NoSuchMethodException: me.lucko.commodore.ReflectionCommodore.removeChild(com.mojang.brigadier.tree.RootCommandNode,java.lang.String)
	at java.lang.Class.getDeclaredMethod(Class.java:2675) ~[?:?]
	at cloud.commandframework.bukkit.CloudCommodoreManager.unregisterWithCommodore(CloudCommodoreManager.java:130) ~[?:?]
	at cloud.commandframework.bukkit.CloudCommodoreManager.unregisterExternal(CloudCommodoreManager.java:86) ~[?:?]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at cloud.commandframework.bukkit.BukkitPluginRegistrationHandler.unregisterRootCommand(BukkitPluginRegistrationHandler.java:163) ~[?:?]
	at cloud.commandframework.CommandManager.deleteRootCommand(CommandManager.java:545) ~[?:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at cloud.commandframework.bukkit.CloudBukkitListener.onPluginDisable(CloudBukkitListener.java:54) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-api-1.18.2-R0.1-SNAPSHOT.jar:?]
	... 12 more
```

</details>

The usage of `removeChild` was introduced here: https://github.com/Incendo/cloud/commit/2572b73c4b81a074facafad1bdfa67a9bcf2277c#diff-28a22aef9a6b4ea00908801a6a0bf2323a926d65beb2eb723ca276edae9fc6f0R120-R125

Which relies on the `CommodoreImpl` class. However, that class was removed/renamed here (which is the cause of this issue): https://github.com/lucko/commodore/commit/98f39247bf37ae5e754589a60bfd0d6f5c35533d

This was released in `commodore` version 2.1, so the issue won't happen on `commodore` <= 2.0 (but newer versions are required for proper 1.19.2 support).

The `removeChild` method was moved to the `AbstractCommodore` class: https://github.com/lucko/commodore/commit/98f39247bf37ae5e754589a60bfd0d6f5c35533d#diff-922a06faa81bdd5d2362da03cf58d617570d3f28757f25ad39303b6e7f56fce2R86-R89

So a "simple fix" could be to update the reflection code to use that class instead, and fall back to the old `CommodoreImpl` class name. I would submit a PR, but I'm short on time right now, and you know the code better than I do. :slightly_smiling_face:

Edit: Or maybe it's better to drop support for `commodore` <= 2.0 and just use the new class name?

**Comment by jpenilla** [comment]
Should be resolved in 1.8.0-SNAPSHOT, testing appreciated

**Comment by 0dinD** [comment]
Works like a charm, thanks for the quick fix!
